readthedocs/storage/s3_storage.py

```diff
@@ -9,16 +9,40 @@
 
 # Disable abstract method because we are not overriding all the methods
 # pylint: disable=abstract-method
+from functools import cached_property
+
 from django.conf import settings
 from django.core.exceptions import ImproperlyConfigured
 from storages.backends.s3boto3 import S3Boto3Storage, S3ManifestStaticStorage
 
 from readthedocs.builds.storage import BuildMediaStorageMixin
+from readthedocs.storage.rclone import RCloneS3Remote
 
 from .mixins import OverrideHostnameMixin, S3PrivateBucketMixin
 
 
-class S3BuildMediaStorage(BuildMediaStorageMixin, OverrideHostnameMixin, S3Boto3Storage):
+class S3BuildMediaStorageMixin(BuildMediaStorageMixin, S3Boto3Storage):
+    @cached_property
+    def _rclone(self):
+        provider = "AWS"
+        # If a custom endpoint URL is given and
+        # we are running in DEBUG mode, use minio as provider.
+        if self.endpoint_url and settings.DEBUG:
+            provider = "minio"
```

Comment:
Shouldn't this be a setting instead of having this hardcoded here? Like `RTD_RCLONE_PROVIDER` so we can define different values depending on the environment?

Reply:
This is already possible by using a different storage backend.

Reply:
The rclone class should be attached to the storage backend chosen, so I don't think there need for a different setting.

Reply:
MinIO is fine. My point is about avoid using `settings.DEBUG` in the code.